audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/stream/SerDesConfig.java

```diff
@@ -132,15 +148,34 @@ private SerDesConfig() {
     /**
      * Kafka value deserialiser for upstream messages coming in as input
      *
+     * @param configProperties contains the directory for error files
      * @return an appropriate value deserialiser for the topic's message content (AuditSuccessMessage)
      */
-    public static Deserializer<AuditSuccessMessage> successValueDeserializer() {
+    public static Deserializer<AuditSuccessMessage> successValueDeserializer(final AuditServiceConfigProperties configProperties) {
         return (String ignored, byte[] auditRequest) -> {
             try {
                 return MAPPER.readValue(auditRequest, AuditSuccessMessage.class);
             } catch (IOException e) {
-                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + new String(auditRequest, Charset.defaultCharset()), e);
+                String failedAuditString = new String(auditRequest, Charset.defaultCharset());
+                createFile("Success-", failedAuditString, configProperties);
+                SerDesHealthIndicator.addSerDesExceptions(e);
+                throw new SerializationFailedException(DESERIALIZATION_FAILED_MESSAGE + failedAuditString, e);
             }
         };
     }
+
+    private static void createFile(final String prefix, final String failedAuditString, final AuditServiceConfigProperties configProperties) {
+        String fileName = prefix + ZonedDateTime.now(ZoneOffset.UTC)
+                .format(DateTimeFormatter.ISO_INSTANT).replace(":", "-");
```

Comment:
might be worth adding a little comment in the code, or in the readme as to why theres a .replace, blame windows

Reply:
Added an inline comment about the replace